source/client/benchmark_client_impl.h

```diff
@@ -50,6 +51,73 @@ class Http1PoolImpl : public Envoy::Http::Http1::ProdConnPoolImpl {
   void createConnections(const uint32_t connection_limit);
 };
 
+// Vanilla Envoy's HTTP/2 pool is single connection only (or actually sometimes dual in connection
+// drainage scenarios). Http2PoolImpl is an experimental pool, which uses multiple vanilla Envoy
+// HTTP/2 pools under the hood. Using multiple connections is useful when testing backends that need
+// multiple connections to distribute load internally. Combining multiple connections with
+// --max-requests-per-connection may help as well, as doing periodically initiating new connections
+// may help the benchmark target by giving it an opportunity to rebalance.
+class Http2PoolImpl : public Envoy::Http::ConnectionPool::Instance,
```

Comment:
This is remarkably simple; I suggest you sync with @antoniovicente about how this might be delivered in Envoy.

Reply:
@antoniovicente do you have any thoughts / comments on this experimental approach to adding support for multiple h2 connections? Maybe we can take this forward as a Nighthawk-specific approach?

Reply:
Sorry, I missed this review until now.  I'll try to look later today.

I wonder what's the status of the proposal at: https://docs.google.com/document/d/1gzY1y8CdbvZZ_82y9Fb6kNXAlzOBC0VIeiuMdn_xCRs/edit

Reply:
@antoniovicente 
That draft certainly looks pretty relevant to this, and https://github.com/envoyproxy/nighthawk/pull/198 also :-)

Reply:
Commented on #198

Reply:
I think that the general Envoy case would benefit from minimizing the number of H2 connections in use.  This implementation looks good for the use case of us having a fairly steady known QPS that would result in good utilization of the backend connection when operating in LRU mode.  One potential worry is connections idling out before they are reused via RR, but that may be a desirable thing in a loadtest scenario.

I pinged in the document, I'ld like multi H2 connection support to be implemented in Envoy core.  Several projects depend on that fix.